.github/workflows/build_and_test.yml

```diff
@@ -110,7 +117,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ ubuntu-latest, windows-latest, macos-latest ]
```

Comment:
Why?

Reply:
We had a duplicated execution on ubuntu: main job (`build and test and upload codecov`) is basically the same, but also with codecov. So there is no need for a separate build_and_test on ubuntu